main.tf

```diff
@@ -33,6 +24,27 @@ resource "azurerm_key_vault" "this" {
       virtual_network_subnet_ids = network_acls.value.virtual_network_subnet_ids
     }
   }
+
+  lifecycle {
+    ignore_changes = [contact] // handled with separate resource to avoid ACL-problems
+  }
+}
+
+resource "azurerm_key_vault_certificate_contacts" "this" {
+  count = length(keys(var.contacts)) > 0 ? 1 : 0
+
+  key_vault_id = azurerm_key_vault.this.id
+
+  dynamic "contact" {
+    for_each = var.contacts
+    content {
+      name  = contact.value.name
+      email = contact.value.email
+      phone = contact.value.phone
+    }
+  }
+
+  depends_on = [azurerm_role_assignment.this]
 }
```

Comment:
We will need to add a delay (sadly) so that the KV can pick up the Azure role assignments.

See here for an example:

https://github.com/Azure/terraform-azurerm-avm-res-keyvault-vault/blob/cd84293ed19e96b4c4e06c2ad777597ab2a2b900/main.secrets.tf#L28C1-L39C2